### ISSUE TITLE
Out of src tree build doesn't work with cmake 2.8.12-2 on ubuntu 14.10

### ISSUE THREAD

**Issue by mgerhardy** [floooh/fips#79]
http://stackoverflow.com/questions/20610255/how-to-tell-cmake-where-to-put-build-files

```
diff --git a/mod/tools/cmake.py b/mod/tools/cmake.py
index 65de979..cd8fea8 100644
--- a/mod/tools/cmake.py
+++ b/mod/tools/cmake.py
@@ -49,7 +49,8 @@ def run_gen(cfg, project_dir, build_dir, toolchain_path, defines) :
             cmdLine += ' -D{}={}'.format(key, cfg['defines'][key])
     for key in defines :
         cmdLine += ' -D{}={}'.format(key, defines[key])
-    cmdLine += ' ' + project_dir
+    cmdLine += ' -B' + build_dir
+    cmdLine += ' -H' + project_dir
     
     res = subprocess.call(cmdLine, cwd=build_dir, shell=True)
     return res == 0
```

**Comment by mgerhardy** [comment]
this even works for me with cmake 3.1.0 on ubuntu 14.10

but without it, it always generates the build files in the src dir - no matter what the working dir for the cmake call is.

**Comment by floooh** [comment]
Strange... I'll try to reproduce the problem in a VM when at home, but the fix seems simple enough to do it your way without the cwd.

**Comment by floooh** [comment]
fixed in pull request #80
